src/lib/Shapes.ts

```diff
@@ -94,7 +94,7 @@ export class Shapes {
 		return new UnionValidator(validators);
 	}
 
-	public array<T>(validator: BaseValidator<T>) {
+	public array<T extends unknown[]>(validator: BaseValidator<T[number]>) {
```

Comment:
What does this even fix? Can we please get unit tests?

Reply:
https://github.com/discordjs/discord.js/pull/8242

Reply:
Should probably still add a test case if you can